test/test_http.py

```diff
@@ -4,11 +4,14 @@
 import os
 import sys
 import unittest
+from contextlib import contextmanager
```

Comment:
Tests need to be completely rewritten.

Main things we need to test:
- RequestHandler Request validation
- RequestHandler actual sending process (and all the underlying http stuff)
- YoutubeDL layer stuff:
   - Mapping ydl params to request handlers, ensuring they are working
   - Cleaning Request of ydl options
- Request, Response, Director, etc.